packages/rust/server/src/models/group.rs

```diff
@@ -0,0 +1,224 @@
+use crate::types::DbPool;
+use std::borrow::Borrow;
+
+use sqlx::postgres::PgRow;
+use sqlx::types::chrono;
+
+use crate::models::user::User;
+
+use sqlx::types::Uuid;
+use sqlx::Row;
+
+pub struct Group {
+    pub id: i32,
+    pub uuid: Uuid,
+    pub group_name: String,
+    pub created: chrono::NaiveDateTime,
+    pub members: Vec<Uuid>,
+}
+
+pub struct UserGroup {
+    pub id: i32,
+    pub user_id: i32,
+    pub group_id: i32,
+    pub role_id: Role,
+    pub created: chrono::NaiveDateTime,
+}
+
+#[derive(Debug, PartialEq)]
+#[repr(u8)]
+pub enum Role {
+    Admin = 0,
+    Member = 1,
+}
+
+impl TryFrom<u8> for Role {
+    type Error = ();
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(Role::Admin),
+            1 => Ok(Role::Member),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(sqlx::FromRow)]
+pub struct UserIDs {
+    pub user_ids: Vec<String>,
+}
+
+#[derive(sqlx::FromRow)]
+pub struct GroupIDs {
+    pub group_ids: Vec<UserGroup>,
+}
+
+impl From<&PgRow> for Group {
+    fn from(row: &PgRow) -> Self {
+        Group {
+            id: row.get("id"),
+            uuid: row.get("uuid"),
+            group_name: row.get("group_name"),
+            created: row.get("created"),
+            members: Vec::new(),
+        }
+    }
+}
+
+impl Group {
+    pub async fn delete_user_groups(db: &DbPool, user_id: i32) -> Result<(), sqlx::Error> {
+        sqlx::query("DELETE FROM user_group WHERE user_id = $1")
+            .bind(user_id)
+            .execute(db)
+            .await?;
+        Ok(())
+    }
+
+    pub async fn role(&self, db: &DbPool, user_id: i32) -> Result<Role, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM user_group WHERE group_id = $1 AND user_id = $2;")
+            .bind(self.id)
+            .bind(user_id)
+            .fetch_one(db)
+            .await?;
+        let role_id: i32 = row.get("role_id");
+        let role_enum = match Role::try_from(role_id as u8) {
+            Ok(role_enum) => role_enum,
+            Err(_) => return Err(sqlx::Error::RowNotFound),
+        };
+        Ok(role_enum)
+    }
+
+    pub async fn members(&self, db: &DbPool) -> Result<Vec<User>, sqlx::Error> {
+        let row = sqlx::query("SELECT * FROM user_group WHERE group_id = $1 ")
+            .bind(self.id)
+            .fetch_all(db)
+            .await?;
+        let mut users_in_group: Vec<User> = vec![];
+        for i in row {
+            let user_to_be_added = User::from_id(db, i.get("id")).await?;
+            users_in_group.push(user_to_be_added);
+        }
+        Ok(users_in_group)
+    }
+
+    pub async fn from_uuid(db: &DbPool, uuid: &Uuid) -> Result<Group, sqlx::Error> {
+        Ok(sqlx::query("SELECT * FROM \"group\" WHERE uuid = $1")
```

Comment:
the group returned from here members is empty

Reply:
fixed